site/src/pages/release-notes/1.18.0.mdx

```diff
@@ -0,0 +1,134 @@
+```java
+GrpcService.builder()
+           .addService(myGrpcService, List.of(LoggingService.newDecorator()))
+           .build();
+```
+
+- Armeria's gRPC users can now use an <type://UnframedGrpcErrorHandler> which provides a
+[richer error model](https://cloud.google.com/apis/design/errors#error_model) for unframed requests.
+
+```java
+Server.builder()
+      .service(GrpcService.builder()
+                          .addService(grpcService)
+                          .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.ofRichJson())
+                          .build());
+```
+
+```json
+{
+	"error": {
+		"code": 2,
+		"message": "A custom message!"
+	}
+}
+```
+
+- Multipart file upload is now available for GraphQL users. #3971 #4199
+```java
+GraphqlService service =
+    GraphqlService.builder()
+...
+                  .runtimeWiring(c -> {
+                      c.scalar(MoreScalars.multipartFile());
+                      DataFetcher<String> upload = environment -> {
+                          MultipartFile multipartFile = environment.getArgument("file");
+                          ...
+                      }
+                      c.type("Multipart", typeWiring -> typeWiring.dataFetcher("upload", upload));
+                  })
+...
+```
+
+- Users can now extract the innermost element of <type://Unwrappable> type objects easily using <type://Unwrappable#unwrapAll()> #4338
+```java
+FooDecorator extends SimpleDecoratingHttpClient {
+...
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
+        Object innermost = unwrapAll();
+        ...
+    }
+}
+```
+
+## 📈 Improvements
+
+- <type://AnnotatedServiceBindingBuilder> and <type://VirtualHostAnnotatedServiceBindingBuilder> now
+extend a single interface. #4205 #4210
+- <type://RequestContext> now extends <type://Unwrappable>. #4308
+
+## 🛠️ Bug fixes
+
+- You no longer see an intermittent <type://IllegalStateException> for gRPC services
+when <type://GrpcServiceBuilder#useBlockingTaskExecutor(boolean)> is enabled. #4367 #4374
+- Requests no longer hang when gRPC-JSON transcoding is enabled and gRPC decorators return an unframed response. #4337 #4348
+- Files from OSGI bundle classloader are correctly served #4371 #4373
+- Server metrics related to a request's start time are now correctly
+recorded when Armeria's server receives the first header. #4365 #4372
+- You no longer see a false positive warning for unknown `MethodType` when using gRPC clients. #4363
+- You no longer see a `NoSuchElementException` while a <type://HealthCheckedEndpointGroup> is closed. #4349
+- <type://ClosedSessionException> and <type://ClosedStreamException> are raised correctly according to its usage. #4380
+- The timeout for client's endpoint selection isn't clamped by `responseTimeout` anymore. #4383
+
+## 🏚️ Deprecations
+
+- N/A
+
+## ☢️ Breaking changes
+
+- N/A
+
+## ⛓ Dependencies
+
+TBU
+
+## 🙇 Thank you
+
+<ThankYou usernames={[
+  'CLAassistant',
```

Comment:
It should be a bug. 😱 😱 😱
```suggestion
```

Reply:
Oh.. 😅   Let me also push a fix for this in the script 🤣